File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& v) {
        int n=v.size(),mx=0,c=1,f=0,m1;
        set<int> s;
        for(auto q:v) s.insert(q);
        for(auto q:s){
            if(f==0){
                f=1;
                m1=q;
                mx++;
            }
            else{
                if(q-m1==1) {
                    c++;
                    mx=max(mx,c);
                }
                else c=1;
                m1=q;
            }
        }
        return mx;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
public:
    int longestConsecutive(vector<int>& v) {
        set<int> st;
        int mx=0,c=1,f=0,x;
        if(v.size()==0) return 0;
        for(auto q:v) st.insert(q);
        for(auto q:st){
            if(f==0){
                f=1;
                x=q;
            }
            else{
                if(q-x==1) c++;
                else {
                    mx=max(mx,c);c=1;
                }
                x=q;
            }
        }
        mx=max(mx,c);
        return mx;
    }
};